content/terraform/v1.14.x (alpha)/docs/language/block/action.mdx

```diff
@@ -0,0 +1,263 @@
+
+The `for_each` argument does not implicitly convert lists or tuples to sets. To declare resource instances based on a nested data structure or combinations of elements from multiple data structures, you can use Terraform expressions and functions to derive a suitable value. Refer to the following examples for more information:
+
+- [Transform a multi-level nested structure into a flat list](/terraform/language/functions/flatten#flattening-nested-structures-for-for_each).
+- [Combine collections to produce a list of element combinations](/terraform/language/functions/setproduct#finding-combinations-for-for_each).
+
+You cannot use sensitive values, such as [sensitive input variables](/terraform/language/block/variable#sensitive), [sensitive outputs](/terraform/language/block/output#sensitive), or [sensitive resource attributes](/terraform/language/expressions/references#sensitive-resource-attributes), as arguments in `for_each`. Becaseu Terraform uses the value in `for_each` to identify the action instance and always discloses it in UI output, sensitive values are not allowed. Terraform returns an error if you attempt to use sensitive values as `for_each` arguments.
+
+If you transform a value containing sensitive data into an argument for use in `for_each`, be aware that most functions in Terraform will return a sensitive result if given an argument with sensitive content. In many cases, you can achieve similar results with a `for` expression. For example, to call `keys(local.map)` where `local.map` is an object with sensitive values, but non-sensitive keys, you can create a value to pass to `for_each` using `toset([for k,v in local.map : k])`.
+
+Refer to [Manage sensitive data](/terraform/language/manage-sensitive-data) for more information.
+
+The `for_each` argument exposes an `each` object that you can reference within the same block to modify specific instances of the action. The object has the following attributes:
+
+- `each.key`: Map key or list member that corresponds to an instance.
+- `each.value`: Map value that corresponds to an instance.
+
+Use the `<TYPE>.<NAME>[<KEY>]` syntax to access an instance of a resource created using `for_each`. For example, `aws_lambda_invoke.server["a_group"]` refers to an invocation of the `aws_lambda_invoke` action named `server` froma key named `a_group`.
+
+The `for_each` argument is a meta-argument, which is built into Terraform and controls the way that Terraform creates resources. Refer to [Meta-arguments](/terraform/language/meta-arguments) for more information.
+
+#### Summary
+
+- Data type: Map or set of strings.
+- Default: None.
+- Example: [Invoke an action multiple times](#invoke-an-action-multiple-times).
+
+### `count`
+
+The `count` meta-argument instructs Terraform to invoke an action multiple times using the same configuration.
```

Comment:
Similar questions as with for_each about whether this happens at "plan" time, allowing users to selectively invoke one or more instances of the action, or whether it happens at "apply" time, meaning Terraform will always invoke the given action "N" times.